API.md

```diff
@@ -0,0 +1,66 @@
+
+### `Octofet(uint8_t pinCS, uint8_t pinMOSI, uint8_t pinSCK, uint8_t devicesCount = 1, uint8_t* resetValues = NULL)`
+
+Constructs a new object of data bus Software SPI with arguments:
+
+- `pinCS`: pin controls to enabled and disabled device on bus. Any GPIO-pin can be used.
+- `pinMOSI`: pin for sending data to the peripherals. Any GPIO-pin can be used.
+- `pinSCK`: pin the clock pulses which synchronize data transmission generated by the master. Any GPIO-pin can be used.
+- `devicesCount`: quantity of octets boards connected in one chain: logical output pin headers of one module to the inputs of another. If the `devicesCount` argument is omitted, it used only one board Octofet.
+- `resetValues`: states power switchers by default as 8-bit value. One bit for one power switch: 1 is turn-on and 0 is turn-off. If `resetValues` argument is adjusted, then when calling methods `begin()` or `reset()`, power switchers sets value by `resetValues`. If `resetValues` argument is omitted — power switchers sets is turn-off.
+
+Difference between Hardware and Software SPI is speed–with software SPI it will run much slower than hardware SPI, but Hardware SPI works only on specific pins: `MOSI`, `MISO`(not used in Octofet) and `SCK`. Use Software SPI only when Hardware SPI pins are used by other devices.
+
+### `void begin()`
+
+Initializes the given interface, prepares the board for communication. Important, call this method at least once, in your `setup()`.
+
+### `digitalWrite(uint8_t device, uint8_t pin, bool value)`
+
+Sets state (turn-on or turn-off) for one power switch. Function arguments:
+
+- `device`: ordinal number of device on the bus. Range is from `1` to `n`, where `n` last number device.
+- `pin`: ordinal number power switch on device. Range is from `0` to `7`.
+- `value`: states selected power switch. Values (turn-on or turn-off) may be switch is different data types: `HIGH` or `LOW` / `true` or `false` / `1` or `0`.
+
+### `digitalWrite(uint8_t pin, bool value)`
+
+Sets state (turn-on or turn-off) for one power switch. Function arguments:
+
+- `pin`: ordinal number power switch on device. Range is from `0` to `7`.
+- `value`: states selected power switch. Values (turn-on or turn-off) may be switch is different data types: `HIGH` or `LOW` / `true` or `false` / `1` or `0`.
+
+A method is a special case of a function `digitalWrite(uint8_t device, uint8_t pin, bool value)`, when there is only one device on the bus.
+
+### `deviceWrite(uint8_t device, uint8_t value)`
+
+Sets state for all power switches. Function arguments:
+
+- `device`: ordinal number of device on the bus. Range is from `1` to `n`, where `n` last number device.
+- `value`: 8-bit value for all of 8 power switchers. One bit for one power switch: 1 is turn-on and 0 is turn-off.
+
+### `deviceWrite(uint8_t value)`
```

Comment:
It might be useful to add a function `getDeviceValue` that returns the current values of the device.

Reply:
@AlexErAmp are you sure we need to return the power switchers states? Why is it for the user?Maybe create a function `toogle`, which inverts the current state power switch?